nginx/nginx.conf

```diff
@@ -0,0 +1,30 @@
+server {
+    listen       80;
+    server_name  localhost;
+
+    set_real_ip_from 128.138.64.0/23;
+    set_real_ip_from 128.138.135.0/24;
+    real_ip_header X-Forwarded-For;
+    real_ip_recursive on;
+
+    location / {
+        root   /usr/share/nginx/html;
+        autoindex on;
+
+        # Only allow whitelisted IPs or basic auth
+        satisfy any;
```

Comment:
"any" here means users who pass any of the auth rules below will be permitted. e.g. if the user passes the `allow`/`deny` rules, then they don't need to pass the `auth_basic` rules -- those are only required for non-whitelisted IPs.

Reply:
I did this in response to Lynn asking for a copy of QGR but not having access to the VM where we host it.